sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PortEvaluationStage.java

```diff
@@ -205,6 +227,20 @@ protected void setProtos(PodInfoBuilder podInfoBuilder, Protos.Resource resource
         return dynamicPort;
     }
 
+    private static Optional<Integer> selectOverlayPort(PodInfoBuilder podInfoBuilder) {
+        // take the next available port in the range.
+        Optional<Integer> dynamicPort = Optional.empty();
+        for (int i = DcosConstants.OVERLAY_DYNAMIC_PORT_RANGE_START;
+             i <= DcosConstants.OVERLAY_DYNAMIC_PORT_RANGE_END; i++) {
+            if (!podInfoBuilder.isAssignedOverlayPort(i)) {
```

Comment:
Looks like there could be a conflict: `isAssignedOverlayPort()` doesn't check for non-dynamic port usage, so if the task is directly reserving e.g. port `1025`...